sympy/functions/elementary/exponential.py

```diff
@@ -1063,17 +1066,24 @@ def mul(d1, d2):
             coeff = -S.NegativeOne**k/k
             for ex in pk:
                 terms[ex] = terms.get(ex, S.Zero) + coeff*pk[ex]
+                terms[ex] = terms[ex].nsimplify()
```

Comment:
Storing twice to a dict (and also loading twice from) is inefficient. A temporary storage is preferable:

                _ = terms.get(ex, S.Zero) + coeff*pk[ex]
                terms[ex] = _.nsimplify()